tests/test_backend_smt.py

```diff
@@ -0,0 +1,491 @@
+
+
+(check-sat)
+'''
+        str_concrete = claripy.StringV("12")
+        solver = self.get_solver()
+        res = claripy.StrToInt(str_concrete, 32)
+        solver.add(res == 12)
+        script = solver.get_smtlib_script_satisfiability()
+        self.assertEqual(correct_script, script)
+
+    def test_str_extract(self):
+        correct_script = '''(set-logic ALL)
+(declare-fun STRING_symb_str_extract () String)
+(assert (let ((.def_0 (= ( str.substr STRING_symb_str_extract 5 1) "abc"))) .def_0))
+(check-sat)
+'''
+        str_symb = claripy.StringS("symb_str_extract", 12, explicit_name=True)
+        res = claripy.StrExtract(0, 1, claripy.StrExtract(1, 2, claripy.StrExtract(4, 8, str_symb)))
+        solver = self.get_solver()
+        solver.add(res == claripy.StringV("abc"))
+        script = solver.get_smtlib_script_satisfiability()
+        # with open("dump_strextract.smt2", "w") as dump_f:
+        #     dump_f.write(script)
+        self.assertEqual(correct_script, script)
+
+    def test_is_digit(self):
+        correct_script = '''(set-logic ALL)
+(declare-fun STRING_symb_str_is_digit () String)
+(assert (let ((.def_0 (= ( str.to.int STRING_symb_str_is_digit ) (- 1)))) (let ((.def_1 (not .def_0))) .def_1)))
+(check-sat)
+'''
+        str_symb = claripy.StringS("symb_str_is_digit", 12, explicit_name=True)
+        res = claripy.StrIsDigit(str_symb)
+        solver = self.get_solver()
+        solver.add(res)
+        script = solver.get_smtlib_script_satisfiability()
+        self.assertEqual(correct_script, script)
+
+    def test_is_digit_simplification(self):
+        correct_script = '''(set-logic ALL)
+
+
+(check-sat)
+'''
+        str_concrete = claripy.StringV("1")
+        res = claripy.StrIsDigit(str_concrete)
+        solver = self.get_solver()
+        solver.add(res)
+        script = solver.get_smtlib_script_satisfiability()
+        self.assertEqual(correct_script, script)
+
+    def test_int_to_str(self):
+        correct_script = '''(set-logic ALL)
+(declare-fun symb_int () Int)
+(assert (let ((.def_0 (= ( int.to.str symb_int ) "12"))) .def_0))
+(check-sat)
+'''
+        int_symb = claripy.BVS("symb_int", 32, explicit_name=True)
+        res = claripy.IntToStr(int_symb)
+        solver = self.get_solver()
+        solver.add(res == claripy.StringV("12"))
+        script = solver.get_smtlib_script_satisfiability()
+        self.assertEqual(correct_script, script)
+
+    def test_int_to_str_simplification(self):
+        correct_script = '''(set-logic ALL)
+
+
+(check-sat)
+'''
+        int_concrete = claripy.BVV(0xc, 32)
+        res = claripy.IntToStr(int_concrete)
+        solver = self.get_solver()
+        solver.add(res == claripy.StringV("12"))
+        script = solver.get_smtlib_script_satisfiability()
+        self.assertEqual(correct_script, script)
+
+if __name__ == "__main__":
+    suite = unittest.TestLoader().loadTestsFromTestCase(TestSMTLibBackend)
+    unittest.TextTestRunner(verbosity=2).run(suite)
```

Comment:
Does the current scheme work with `nosetests`? Note that our CI is built on top of nosetests.

Reply:
These test currently works out of the box with `nosetests` on my machine

Reply:
Awesome!